build_tools/upload_logs_to_s3.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+import os
+import glob
+import subprocess
+
+
+def run_aws_cp(src, dest, content_type=None):
+    cmd = (
+        ["aws", "s3", "cp", src, dest, "--recursive"]
+        if os.path.isdir(src)
+        else ["aws", "s3", "cp", src, dest]
+    )
+    if content_type:
+        cmd += ["--content-type", content_type]
+    try:
+        print(f"[INFO] Running: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"[ERROR] Failed to upload {src} to {dest}: {e}")
+
+
+def upload_logs_to_s3(bucket_name, run_id, amdgpu_family, log_dir="build/logs"):
+    s3_base_path = f"s3://{bucket_name}/{run_id}/logs/{amdgpu_family}"
+
+    if not os.path.isdir(log_dir):
+        print("[INFO] No logs directory found. Creating it.")
+        os.makedirs(log_dir)
+        return
+
+    log_files = glob.glob(os.path.join(log_dir, "*.log"))
+    if not log_files:
+        print("[WARN] No .log files found. Skipping log upload.")
+    else:
+        run_aws_cp(log_dir, s3_base_path, content_type="text/plain")
+
+
+if __name__ == "__main__":
+    bucket = os.getenv("S3_BUCKET", "therock-artifacts")
+    run_id = os.getenv("GITHUB_RUN_ID")
+    amdgpu_family = os.getenv("AMDGPU_FAMILIES", "unknown")
```

Comment:
Should we rather let it fail instead of pushing to a default location? I am undecided here if it makes sense to upload logs in that case at all as it rather is an CI issue than a build issue if the variable is not set correctly at all.

Reply:
With the above, `amdgpu_family` defaults to "unknown" and thus `not amdgpu_family` will never catch that is unset.

Reply:
Forgot that part - updated